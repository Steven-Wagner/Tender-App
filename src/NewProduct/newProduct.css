.new-product-form {
    display: flex;
    flex-direction: column;
    margin: 0 20px;
}

.price-ad-wrapper {
    display: flex;
    flex-direction: column;
}

.price-wrapper {
    display: flex;
    justify-content: flex-start;
}

.price-wrapper input {
    max-width: 100px;
    border-radius: 5px;
}

.new-product label, 
.new-product-form input, 
.new-product-form textarea, 
.new-product-form select {
    margin: 20px 0;
    padding: 10px;
    border-radius: 5px;
}

#title {
    font-size: 1.5em;
}

#img, #description, #price {
    font-size: 1em;
}

#price {
    margin: 0 0 0 10px;
}

.submit-new-product-btn {
    margin: 20px;
    width: 50%;
    font-size: 1.5em;
    align-self: center;
}

#advertise {
    margin: 10px;
    padding: 5px;
    width: 90%;
    max-width: 650px;
    align-self: center;
    font-size: 1.25em;
    white-space: initial;
}

.current-advertising-label {
    margin-top: 20px;
    font-weight: bold;
}