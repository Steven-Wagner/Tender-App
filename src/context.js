import React from 'react';

const TenderContext = React.createContext();

export const ProductsProvider = TenderContext.Provider
export const ProductsConsumer = TenderContext.Consumer

export default TenderContext;