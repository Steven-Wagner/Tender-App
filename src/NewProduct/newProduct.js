import React, {Component} from 'react';
import Nav from '../Components/Nav/nav';
import './newProduct.css';
import TenderContext from '../context';

class NewProduct extends Component {

    static contextType = TenderContext;

    constructor(props) {
        super(props);
        this.state = {
            item: {
                title: '',
                img: '',
                description: '',
                price: '',
                advertise: 'None'
            }
        }
        this.handleChangeInput = this.handleChangeInput.bind(this);
        this.validateNewProduct = this.validateNewProduct.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
    }

    handleChangeInput(e) {
        const newItem = Object.assign({}, this.state.item);
        newItem[e.target.id] = e.target.value;
        this.setState({
            item: newItem
        });
    }

    handleSubmit(e) {
        e.preventDefault();
        
        const validate = this.validateNewProduct(this.state.item);

        if (validate) {
            this.context.setPopupMessages('popup', 'New Product Created!')
            this.resetValues();
            // document.getElementById('productForm').reset();
        }
    }

    resetValues() {
        this.setState({
            item: {
                title: '',
                img: '',
                description: '',
                price: '',
                advertise: 'None'
            }
        })
    }

    validateNewProduct(item) {
        let errorMessages = [];

        if (!item.title) {
            errorMessages.push('Item must have a title');
        }

        if (isNaN(parseFloat(item.price)) || parseFloat(item.price) < 1) {
            errorMessages.push('Item price must be at least 1 Play Money')
        }

        if (errorMessages.length > 0) {
            this.context.setPopupMessages('errorPopup', errorMessages)
            return false;
        }
        else {return true}
    }

    render() {
        return(
            <div>
                <Nav currentComponent={'NewProduct'}/>
                <header>
                    <h2 className="page-header">New Product</h2>
                </header>
                <form id='productForm' className="new-product-form" onSubmit={(e) => this.handleSubmit(e)}>
                    <label htmlFor="title">Title</label>
                    <input 
                        value={this.state.item.title}
                        type="text" 
                        id="title"
                        onChange={(e) => this.handleChangeInput(e)}/>
                    
                    <label htmlFor="img">Image URL (optinal)</label>
                    <input 
                        value={this.state.item.img}
                        type="text" 
                        id="img"
                        onChange={(e) => this.handleChangeInput(e)}/>

                    <label htmlFor="description">Product Description</label>
                    <textarea 
                        value={this.state.item.description}
                        id="description" 
                        placeholder='Machine washable with like colors'
                        onChange={(e) => this.handleChangeInput(e)}>
                    </textarea>

                    <div className="price-ad-wrapper">
                        <div className="price-wrapper">
                            <label className="price" htmlFor="price">Price:</label>
                            <input 
                                value={this.state.item.price}
                                id="price" 
                                type="number" 
                                step="1"
                                onChange={(e) => this.handleChangeInput(e)}/>
                        </div>
                        
                        <label className="current-advertising-label" htmlFor="advertise">Ad Spending</label>
                        <select 
                            id="advertise" 
                            onChange={(e) => this.handleChangeInput(e)}
                            value={this.state.item.advertise}>
                            <option value='None'>None</option>
                            <option value='Homepage ads'>Homepage ads - 50 Play Money per day</option>
                            <option value='Popup ads'>Popup ads - 100 Play Money per day</option>
                            <option value='Annoying Ads'>Annoying Ads - 150 Play Money per day</option>
                        </select>
                    </div>

                    <button className="submit-new-product-btn" type='submit'>Submit</button>

                </form>
            </div>
        )
    }
}

export default NewProduct;