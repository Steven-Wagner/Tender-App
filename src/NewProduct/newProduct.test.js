import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import NewProduct from './newProduct';
import Enzyme, { shallow, mount, render, setState, update } from 'enzyme';
import Adapter from "enzyme-adapter-react-16";
import {ProductsProvider} from '../context';
import ReactDOM from 'react-dom';

NewProduct.prototype.componentDidMount = () => {};

Enzyme.configure({ adapter: new Adapter() });

jest.mock('../services/Token-services');

function makeNewProductWrapper() {
    return mount(
            <Router>
                <ProductsProvider value={context}>
                    <NewProduct/>
                </ProductsProvider>
            </Router>)
}

const context = {
    userInfo: {
        username: 'Test User',
        money: '150.00'
    }
};

describe('NewProduct UI renders correctly', () => {
    it('renders content without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(
            <Router>
                <ProductsProvider value={context}>
                    <NewProduct/>
                </ProductsProvider>
            </Router> , div);
    });
      
    it('NewProduct renders correctly', () => {
        const tree = makeNewProductWrapper();
        expect(tree).toMatchSnapshot();
    });
})