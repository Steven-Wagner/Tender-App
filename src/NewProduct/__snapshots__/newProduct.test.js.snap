// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewProduct UI renders correctly NewProduct renders correctly 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <NewProduct>
      <div
        className="nav-space"
      >
        <Nav
          currentComponent="NewProduct"
        />
        <div
          className="page-wrapper"
        >
          <header>
            <h2
              className="page-header"
            >
              New Product
            </h2>
          </header>
          <form
            className="new-product-form"
            id="productForm"
            onSubmit={[Function]}
          >
            <label
              htmlFor="title"
            >
              Title
            </label>
            <input
              id="title"
              onChange={[Function]}
              type="text"
              value=""
            />
            <label
              htmlFor="img"
            >
              Image URL (optinal)
            </label>
            <input
              id="img"
              onChange={[Function]}
              type="url"
              value=""
            />
            <label
              htmlFor="description"
            >
              Product Description
            </label>
            <textarea
              className="new-description"
              id="description"
              onChange={[Function]}
              placeholder="Machine washable with like colors"
              value=""
            />
            <div
              className="price-ad-wrapper"
            >
              <div
                className="price-wrapper"
              >
                <label
                  className="price"
                  htmlFor="price"
                >
                  Price:
                </label>
                <input
                  id="price"
                  onChange={[Function]}
                  step="1"
                  type="number"
                  value=""
                />
              </div>
              <label
                className="current-advertising-label"
                htmlFor="advertise"
              >
                Ad Spending
              </label>
              <select
                className="select-ad"
                id="ad"
                onChange={[Function]}
              >
                <option
                  value="None"
                >
                  None
                </option>
                <option
                  value="Homepage ads"
                >
                  Homepage ads - 
                  0
                   Play Money per day
                </option>
                <option
                  value="Popup ads"
                >
                  Popup ads - 
                  0
                   Play Money per day
                </option>
                <option
                  value="Annoying ads"
                >
                  Annoying Ads - 
                  0
                   Play Money per day
                </option>
              </select>
            </div>
            <button
              className="submit-new-product-btn"
              type="submit"
            >
              Submit
            </button>
          </form>
        </div>
      </div>
    </NewProduct>
  </Router>
</BrowserRouter>
`;
